src/utils.h

```diff
@@ -16,6 +16,9 @@
 //! Test if string a and b are the same case-insensitively
 #define STR_CASEQ(a, b) (strcasecmp(a, b) == 0)
 
+//! Convert a second into the internal time representation
+#define SECS(s) (((gint64)s) * 1000 * 1000)
```

Comment:
Why not use a function?

```c
gint64 seconds2us(gint64 seconds) {
       // do some overflow detection as a bonus
       return seconds * 1000 * 1000;
}
```

If performance is an issue it can also go in the header as `static inline`.

Reply:
Ok.

```
       // do some overflow detection as a bonus
```

Explain me, where do you do overflow detection here?

Reply:
I don't in that example! It was a suggestion as an upside to using a function. Though I'm not sure the chances of a 64 bit int overflowing here.

At least the upside for me is that it saves the macro parenthesis madness.

Reply:
Well, I'd guess even just making the `gint64 seconds` const will do the job like compilers would optimize `strlen("asdf")` to its output.

It think it's a perfect opportunity to test the optimisation levels and look into its binary output.

Reply:
Ok, I researched it. Starting with `-O2` it gets optimized. Fine. As expected. Changed it.

Compiled:

```
src/../config.h:15:15: error: initializer element is not constant
 .timeouts = { s2us(10), s2us(10), s2us(0) }, /* low, normal, critical */
               ^~~~
```

:see_no_evil: 

Well, I guess, there is no discussion necessary, if we want a function, right? :grin: